htsget-test/src/util.rs

```diff
@@ -18,6 +18,24 @@ pub fn generate_test_certificates<P: AsRef<Path>>(
   (key_path, cert_path)
 }
 
+pub fn generate_key_pair<P: AsRef<Path>>(
+  in_path: P,
+  private_key: &str,
+  public_key: &str,
+) -> (Vec<u8>, Vec<u8>) {
+  let private_key = in_path.as_ref().join(private_key);
+  let public_key = in_path.as_ref().join(public_key);
+
+  let key_pair = KeyPair::generate().unwrap();
+  let private_key_pem = key_pair.serialize_pem();
+  let public_key_pem = key_pair.public_key_pem();
+
+  fs::write(&private_key, &private_key_pem).unwrap();
```

Comment:
Would be interesting to have all this generated in-memory and only serialised to disk when and if needed? This helper assumes it needs to be on disk and therefore all downstream code will tag along that assumption... I understand that it can be a ton of work to get this interoperating well with all the other systems, just thinking out loud here...

Reply:
Yeah that's true, actually this one doesn't need to be written at all because the config file takes as input a `Vec<u8>`.